FluentAssertions.Core/TypeInfo.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Reflection;
+
+namespace FluentAssertions
+{
+#if NO_TYPEINFO
+
+	internal sealed class TypeInfo
+	{
+		private readonly Type type;
+		internal TypeInfo(Type type)
+		{
+			this.type = type;
+		}
+
+		public Assembly Assembly { get { return type.Assembly; } }
+
+		public CustomAttributeData[] CustomAttributes
+		{
+			get
+			{
+#if CUSTOM_ATTRIBUTEDATA
```

Comment:
If this isn't defined then you won't have a return value for the get. Therefore no compilation.